### DISCUSSION TITLE
8 bars patterns and randomization/probability

### DISCUSSION THREAD

**Discussion by Piergap** [bangcorrupt/hacktribe#81] [Ideas]
Hi,
Thanks for all the efforts this community is putting on revitalising the beloved ES2.

I'm surprised no one yet proposed to increase the patterns length from 4 to 8 bars.
As well introducing:
- Probability in steps playback
- automatic selectivebars playback (only play a part on certain bars of the pattern)
Ideas coming from the Pajen-hack for the Volca Sample!
Hopes this gives some inspiration!
Paolo

**Comment by bangcorrupt** [comment]
These would be cool features.

**Comment by juanjo1138** [comment]
Personally, the pattern chain is enough for me and I would not need a greater length of the patterns, but

• Probability factor per step would be great!

• Being able to select the start step in the pattern, just like the end step is selected, would be great, not only for jamming, but for working with the sequencer.

**Reply by bangcorrupt** [reply]
> select the start step in the pattern

This would be great for copy and paste.

**Reply by juanjo1138** [reply]
If it can be done in the same pattern, it can be used to move the sequence…

**Reply by bangcorrupt** [reply]
Copy and paste steps will be added as an external script, keep an eye on [e2-scripts#3](https://github.com/bangcorrupt/e2-scripts/issues/3) and #47. 

There is also a script to [rotate sequences](https://github.com/bangcorrupt/e2-scripts/blob/master/e2seqrot.py) in a pattern file, but it's not integrated with sysex yet.

**Reply by Piergap** [reply]
Though pattern chain would initiate the set of mute-unmute parts each time, which would interfere with live playing. 8 bars length would give more room for sequence creation allowing freedom on live "parts mute-unmute" performing

**Reply by bangcorrupt** [reply]
I haven't looked at longer patterns yet.  It might be as simple as increasing the iterator and giving it more data to read, but I doubt it.  Having mutes carry over between pattern changes would be cool, I don't think it would be too hard to implement.  Just store the mute state in global settings instead of the pattern file, maybe. I managed to get notes to hang when changing patterns at one point, but not in any stable reproducible way.

**Comment by bangcorrupt** [comment]
> Probability in steps playback

> Probability factor per step

See #87.

**Reply by juanjo1138** [reply]
Yes, I have wanted to set it aside in a separate petition :)
